lib/pages/settings.dart

```diff
@@ -58,14 +52,14 @@ class _SettingsPageState extends State<SettingsPage> {
   void initState() {
     super.initState();
 
-    // When closing settings, also shrink expanded setting.
-    widget.isSettingsOpenNotifier.addListener(() {
-      if (!widget.isSettingsOpenNotifier.value) {
-        setState(() {
-          expandedSettingId = null;
-        });
-      }
-    });
```

Comment:
If we now open the locale list for example in the settings, close the settings panel, and then reopen it the locale list will still be open. In the previous version, we closed the settings lists when the panel was closed.

Do we improve the performance by removing this?

Reply:
Calling setState in the original way here was affecting performance (probably because setState is being called at the beginning of the position animation that is already going to call seState. I changed it to hide after the reverse animation completed instead, and this way the functionality is preserved while maintaining the performance when closing the panel.